x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/EqlSession.java

```diff
@@ -27,22 +29,22 @@
 public class EqlSession {
 
     private final Client client;
-    private final Configuration configuration;
+    private final EqlConfiguration configuration;
     private final IndexResolver indexResolver;
 
     private final PreAnalyzer preAnalyzer;
     private final Analyzer analyzer;
     private final Optimizer optimizer;
     private final Planner planner;
 
-    public EqlSession(Client client, Configuration cfg, IndexResolver indexResolver, PreAnalyzer preAnalyzer, Analyzer analyzer,
-            Optimizer optimizer, Planner planner, PlanExecutor planExecutor) {
+    public EqlSession(Client client, EqlConfiguration cfg, IndexResolver indexResolver, PreAnalyzer preAnalyzer,
+            FunctionRegistry functionRegistry, Optimizer optimizer, Planner planner, PlanExecutor planExecutor) {
 
         this.client = new ParentTaskAssigningClient(client, cfg.getTaskId());
         this.configuration = cfg;
         this.indexResolver = indexResolver;
         this.preAnalyzer = preAnalyzer;
-        this.analyzer = analyzer;
+        this.analyzer = new Analyzer(cfg, functionRegistry, new Verifier());
```

Comment:
That's unfortunate - do we need a Verifier instance each time?

Reply:
You are correct. We can create the Verifier _once_ in the plan executor and pass that along.